nixos/modules/services/monitoring/prometheus/default.nix

```
@@ -522,6 +522,45 @@ in {
       '';
     };
 
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      example = "/root/prometheus.env";
+      description = ''
+        Environment file as defined in <citerefentry>
+        <refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum>
+        </citerefentry>.
+
+        Secrets may be passed to the service without adding them to the
+        world-readable Nix store, by specifying placeholder variables as
+        the option value in Nix and setting these variables accordingly in the
+        environment file.
+
+        Environment variables from this file will be interpolated into the
+        config file using envsubst with this syntax:
+        <literal>$ENVIRONMENT ''${VARIABLE}</literal>
+
+        <programlisting>
+          # Example scrape config entry handling an OAuth bearer token
+          {
+            job_name = "home_assistant";
+            metrics_path = "/api/prometheus";
+            scheme = "https";
+            bearer_token = "\${HOME_ASSISTANT_BEARER_TOKEN}";
```

OfBorg failed becaue this variable needs to be escaped with two single quotes in a multi-line string:
```suggestion
            bearer_token = "''${HOME_ASSISTANT_BEARER_TOKEN}";
```
(see the part about multi-line strings in https://nixos.org/manual/nix/stable/#idm140737322059824)

---

I *think* I need `"\''$`. I copied it from my config and envsubst needs to see the literal $ and it's not a '' string in that config. Will test that. :)

---

Oh yeah you're right, the backslash needs to be in the rendered example.